src/update/verify.go

```diff
@@ -3,56 +3,60 @@ package update
 import (
 	"bufio"
 	"bytes"
+	"crypto"
 	"crypto/sha256"
 	_ "embed" // needed for //go:embed
 	"encoding/hex"
 	"fmt"
 	"io"
 	"strings"
 
-	"github.com/ProtonMail/go-crypto/openpgp"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	"github.com/sigstore/sigstore/pkg/signature"
 
 	"github.com/thought-machine/please/src/cli"
 )
 
-// identity is the signing identity of this key.
-const identity = "Please Releases <releases@please.build>"
-
 // pubkey is the public key we verify Please releases with.
-//go:embed pubkey.pem
-var pubkey string
+//
+//go:embed key.pub
+var key []byte
 
 // verifySignature verifies an OpenPGP detached signature of a file.
 // It returns true if the signature is correct according to our key.
-func verifySignature(signed, signature io.Reader) bool {
-	entities, err := openpgp.ReadArmoredKeyRing(strings.NewReader(pubkey))
+func verifySignature(signed, sig io.Reader) bool {
+	return verifySignatureWithKey(signed, sig, key)
+}
+
+func verifySignatureWithKey(signed, sig io.Reader, key []byte) bool {
+	priv, err := cryptoutils.UnmarshalPEMToPublicKey(key)
```

Comment:
Why `priv`? Doesn't this return a public key?

Reply:
Ah, I think I just copied from an example.

Reply:
We should change this to pub then, since this does seem to return a public key: https://pkg.go.dev/github.com/sigstore/sigstore/pkg/cryptoutils#UnmarshalPEMToPublicKey

Reply:
Yup sorry, I hadn't pushed